### PR TITLE
Add body field to Page and use it

### PR DESCRIPTION
**Pull request by XiangRongLin**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [ ] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [ ] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

Is based on #598 so that i can see if it runs in CI

### CODE REVIEW

**Review by TiA4f8R — COMMENTED** [review]
Can you also switch the API domain from `https://www.youtube.com/youtubei/v1` to `https://youtubei.googleapis.com/youtubei/v1` for the POST requests? It can reduce the consent cookies redirects and responses are the same.

File: extractor/src/main/java/org/schabi/newpipe/extractor/Page.java [hunk 1]
```diff
@@ -59,4 +66,8 @@ public static boolean isValid(final Page page) {
         return page != null && (!isNullOrEmpty(page.getUrl())
                 || !isNullOrEmpty(page.getIds()));
     }
+
+    public byte[] getBody() {
```
Comment (TobiGr) [inline extractor/src/main/java/org/schabi/newpipe/extractor/Page.java:70]: Please annotate that method as nullable

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java [hunk 2]
```diff
@@ -259,9 +245,16 @@ private Page getNextPageFrom(final JsonArray contents) throws IOException, Extra
                     .getObject("continuationEndpoint")
                     .getObject("continuationCommand")
                     .getString("token");
+
```
Comment (TiA4f8R) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java:248]: Remove this empty line.

**Review by TiA4f8R — APPROVED** [review]
Looks good, just a nit below.

**Comment by XiangRongLin** [comment]
Test failure is from soundcloud and not from youtube

**Pull request merged**